src/pip/_internal/operations/generate_metadata.py

```diff
@@ -1,12 +1,19 @@
 """Metadata generation logic for source distributions.
 """
 
+import logging
+import os
+
+from pip._internal.utils.misc import call_subprocess, ensure_dir
+from pip._internal.utils.setuptools_build import make_setuptools_shim_args
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
-    from typing import Callable
+    from typing import Callable, List
     from pip._internal.req.req_install import InstallRequirement
```

Comment:
I would put an empty line above this, it can be rolled into the next one if you like.